.travis.yml

```diff
@@ -7,117 +7,134 @@ env:
+      conda info -a;
     fi
-  - bash miniconda.sh -b -p $HOME/miniconda
-  - export PATH="$HOME/miniconda/bin:$PATH"
-  - hash -r
-  - conda config --set always_yes yes --set changeps1 no
-  - conda update -q conda
-  # Useful for debugging any issues with conda
-  - conda info -a
 
 install:
   # Install devito with conda
-  - if [[ $INSTALL_TYPE == 'conda' ]]; then
+  - if [[ ${INSTALL_TYPE} == 'conda' ]]; then
       conda env create -q -f environment.yml python=$TRAVIS_PYTHON_VERSION;
       source activate devito;
       pip install -e .;
       conda list;
     fi
   # Install devito with pip
-  - if [[ $INSTALL_TYPE == 'pip_setup' ]]; then python setup.py install; fi
+  - if [[ ${INSTALL_TYPE} == 'pip_setup' ]]; then python setup.py install; fi
 
 before_script:
   - echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
-  - if [[ $DEVITO_BACKEND == 'yask' ]]; then
+  - if [[ ${DEVITO_BACKEND} == 'yask' ]]; then
       conda install swig; cd ../;
       git clone https://github.com/opesci/yask.git;
       cd yask; make compiler && make compiler-api; pip install -e .; cd ../devito;
     fi
 
 script:
-  - flake8 --builtins=ArgumentError .
-  # Run tests with pip
-  - if [[ $INSTALL_TYPE == 'pip_setup' ]]; then python setup.py test; fi
-  # Run test with py.tests
-  - if [[ $INSTALL_TYPE == 'conda' ]]; then py.test --cov devito tests/; fi
+  - if [[ ${TEST_FLAKE} == 1 ]]; then flake8 --builtins=ArgumentError .; fi
+  # Run tests
+  - if [[ ${TEST_BUNDLE} == 'basic_pip' ]]; then python setup.py test; fi
+  - if [[ ${TEST_BUNDLE} == 'basic_conda' ]]; then py.test --cov devito tests; fi
   # Run remaining specialized examples and tests
   # Additional seismic operator tests
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/benchmark.py test -P acoustic -a; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/acoustic/acoustic_example.py --full; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/acoustic/acoustic_example.py --full --checkpointing; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/acoustic/acoustic_example.py --constant --full; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/misc/linalg.py mat-vec mat-mat-sum transpose-mat-vec; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/tti/tti_example.py -a; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then python examples/seismic/tti/tti_example.py -a --noazimuth; fi
-  # Test tutorial notebooks for the website using nbval
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then py.test --nbval examples/cfd; fi
-  - if [[ $RUN_EXAMPLES == 'True' ]]; then py.test --nbval examples/seismic/tutorials/0[1-3]*; fi
+  - if [[ ${TEST_BUNDLE} == 'tutorials' ]]; then
+    python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5;
```

Comment:
we need one ``if`` on each line, otherwise failing tests won't be caught by Travis. I'm quite sure because we made this same mistake in the past already

Reply:
fixed in cb4eafb